### PR TITLE
Warn if GOPROXY does not support /catalog

### PR DESCRIPTION
**Pull request by zeeke**
Just because it tooks me an hour to understand why it does not show my private repositories. :slightly_smiling_face:

### CODE REVIEW

**Comment by marwan-at-work** [comment]
@zeeke hah! Sorry for driving you into a rabbit hole of my hack-y code. But, thank you very much for the contribution!

File: main.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ func home(fs http.FileSystem) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		mods, err := getCatalogModules(r.Context())
 		if err != nil {
+			fmt.Printf("Error while retrieving catalog from proxy: [%s]\nFallback to publick index", err)
```
Comment (marwan-at-work) [inline main.go]: Typo: `public` not `publick`

Comment (zeeke) [reply main.go]: Sure

**Comment by zeeke** [comment]
> @zeeke hah! Sorry for driving you into a rabbit hole of my hack-y code. But, thank you very much for the contribution!

Very nice work. Go environment miss a tool like this to manage documentation. Hope to be able to contribute with more useful PRs

**Comment by marwan-at-work** [comment]
@zeeke please do! Feel free to reach out to me on the Gopher Slack for any questions.

**Pull request merged**